app/deliveryoptions/reader_type.py

```diff
@@ -0,0 +1,197 @@
+import logging
+import os
+from ipaddress import IPv4Address, IPv6Address, ip_address, ip_network
+from typing import Dict, List, Optional, Union
+
+from app.deliveryoptions.constants import (
+    IP_ONSITE_RANGES,
+    IP_STAFFIN_RANGES,
+    Reader,
+)
+from django.conf import settings
+from django.http import HttpRequest
+
+logger = logging.getLogger(__name__)
+
+
+def is_onsite(visitor_ip_address: str) -> bool:
+    """
+    Check if a visitor's IP address is within the on-site IP ranges.
+
+    Args:
+        visitor_ip_address: The visitor's IP address
+
+    Returns:
+        True if the visitor is on-site, False otherwise
+    """
+    return is_ip_in_cidr(visitor_ip_address, IP_ONSITE_RANGES)
+
+
+def is_subscribed() -> bool:
+    """
+    Check if the user has a subscription.
+
+    Returns:
+        True if the user has a subscription, False otherwise
+
+    TODO: once user management is in place
+    """
+    return False
+
+
+def is_staff(visitor_ip_address: str) -> bool:
+    """
+    Check if a visitor's IP address is within the staff IP ranges.
+
+    Args:
+        visitor_ip_address: The visitor's IP address
+
+    Returns:
+        True if the visitor is staff, False otherwise
+    """
+    return is_ip_in_cidr(visitor_ip_address, IP_STAFFIN_RANGES)
+
+
+def get_reader_type(request: HttpRequest) -> Reader:
+    """
+    Determine the reader type based on request information.
+
+    Args:
+        request: The HTTP request
+
+    Returns:
+        The determined reader type
+    """
+    reader = Reader.UNDEFINED
+
+    try:
+        visitor_ip_address = get_client_ip(request)
+    except Exception as e:
+        logger.warning(
+            f"Cannot determine the users ip address - returning OFFSITE ({type(e)}: {e.args})"
+        )
+        return Reader.OFFSITE
+
+    if reader == Reader.UNDEFINED:
+        if is_subscribed():
+            reader = Reader.SUBSCRIPTION
+        elif is_onsite(visitor_ip_address):
+            reader = Reader.ONSITEPUBLIC
+        elif is_staff(visitor_ip_address):
+            reader = Reader.STAFFIN
+        else:
+            reader = Reader.OFFSITE
+
+    return reader
+
+
+def get_client_ip(request: HttpRequest) -> Optional[str]:
+    """
+    Extract the client IP address from the HTTP request in a secure manner.
```

Comment:
Is `get_client_ip` function going to be validated by those responsible for the settings in this PR?